### PR TITLE
Implement an idea about guarding from obsolete mutations

### PR DESCRIPTION
**Pull request by andon**
I will not merge it, just let me know what you think, ping @ognen

### CODE REVIEW

File: src/read/elements/loading.js [hunk 1]
```diff
@@ -23,8 +23,8 @@ class Loading extends React.Component {
   }
 
   componentDidMount() {
-    const { dispatch, kind, uri, where } = this.props;
-    dispatch({type: 'READ_PERFORM', uri, where, kind});
+    const { dispatch, kind, uri, where, random } = this.props;
```
Comment (ognen) [inline src/read/elements/loading.js]: a) better naming please (readTag, readId, or something similar).

b) random doesn't guarantee uniqueness  (random numbers can repeat, theoretically). Use uuid instead.

File: src/read/reducer.js [hunk 2]
```diff
@@ -19,11 +19,31 @@ import { ReadFn } from './readRegistry';
...
+  if(!element) {
+    // the path for the action can't be accessed in the latest cursor
+    // cursor has changed, so we can only discard the action
+    return cursor;
+  }
+  const canonicalKind = canonical(element.get('kind'));
   const pathToKind = List.of(...action.fromPath, 'kind');
+  if (action.random) {
+    const pathToRandom = List.of(...action.fromPath, 'random');
+    if (action.random !== cursor.getIn(pathToRandom)) {
```
Comment (ognen) [inline src/read/reducer.js]: This should happen only in the READ_SUCCEEDED / READ_FAILED branches of the reducer.

File: src/read/elements/loading.js (hunk 1 above)
Comment (andon) [inline src/read/elements/loading.js]: Yes, this was the plan for both. Will fix.

File: src/read/reducer.js (hunk 2 above)
Comment (andon) [inline src/read/reducer.js]: Initially I only did it for those 2 branches. The thing is, there is another case (the one before), where we discard the changes. But you're right, it should be more clear that for those 2 we are checking it.

Comment (ognen) [inline src/read/reducer.js]: Hm, if you mean to protect from doubly triggering a read, it makes sense.

**Comment by andon** [comment]
Closed this one, since it was done in master and only as an idea. Changes as discussed will be committed to develop.

**Pull request closed without merging**